src/Controller/CRUDController.php

```diff
@@ -1359,15 +1361,51 @@ protected function handleXmlHttpRequestErrorResponse(Request $request, FormInter
             return $this->renderJson([], Response::HTTP_NOT_ACCEPTABLE);
         }
 
-        $errors = [];
-        foreach ($form->getErrors(true) as $error) {
-            $errors[] = $error->getMessage();
+        return $this->json(
+            $this->buildConstraintViolationList($form),
+            Response::HTTP_BAD_REQUEST
+        );
+    }
+
+    private function buildConstraintViolationList(FormInterface $form): ConstraintViolationList
+    {
+        $closure = \Closure::bind(function (ConstraintViolation $violation, string $newPropertyPath): ConstraintViolation {
```

Comment:
This strategy to modify the property path is a bit clunky, it works, but not sure if this is a good idea or I should create new ConstraintViolation and copy the data. wdyt @VincentLanglet ?

Psalm complains and I had to suppress

Reply:
If it's possible to create a new one with different data, might be better indeed.

Reply:
Check new commit, I removed the closure in favour of creating a new object.